### PR TITLE
WIP: Cli rest messages

### PR DESCRIPTION
**Pull request by prestwich**

### CODE REVIEW

File: golang/x/relay/client/cli/tx.go [hunk 1]
```diff
@@ -23,31 +25,62 @@ func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
...
 	return relayTxCmd
 }
 
-// GetCmdSetLink is the CLI command for sending a SetLink transaction
-func GetCmdSetLink(cdc *codec.Codec) *cobra.Command {
+// GetCmdIngestHeaderChain creates a CLI command to ingest a header chain
+func GetCmdIngestHeaderChain(cdc *codec.Codec) *cobra.Command {
 	return &cobra.Command{
-		Use:   "setlink [hex header] [signer address]",
+		Use:   "ingestheaders <json list of headers>",
```
Comment (ErinHales) [inline golang/x/relay/client/cli/tx.go]: `ingestheaderchain`

File: golang/x/relay/client/cli/tx.go [hunk 2]
```diff
@@ -23,31 +25,62 @@ func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
...
+		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cliCtx := context.NewCLIContext().WithCodec(cdc)
 
 			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
 
-			addr := sdk.AccAddress(args[1])
-			msg := types.NewMsgSetLink(addr, args[0], cliCtx.GetFromAddress())
+			// note: unmarshalling json here is undesirable, but necessary
+			// 	     for MarkNewHeaviest we want to accept text instead of json
```
Comment (ErinHales) [inline golang/x/relay/client/cli/tx.go]: `for IngestHeaderChain`

Comment (prestwich) [reply golang/x/relay/client/cli/tx.go]: no, this is an informational note about the future :)

File: golang/x/relay/client/cli/tx.go (hunk 1 above)
Comment (prestwich) [reply golang/x/relay/client/cli/tx.go]: in the interest of being shorter to type in a cli, might leave it like this. next line needs fixing though

Comment (ErinHales) [reply golang/x/relay/client/cli/tx.go]: My main concern is there is an `ingestHeaderChain` function and an `ingestHeaders` function.  So calling this `ingestHeaders` might be confusing

Comment (prestwich) [reply golang/x/relay/client/cli/tx.go]: eh, `ingestHeaders` is internal only. it's never part of the outside interface

Comment (ErinHales) [reply golang/x/relay/client/cli/tx.go]: Ok, that's fine then

**Pull request merged**